client/my-sites/sites/index.jsx

```diff
@@ -24,12 +24,12 @@ export class Sites extends Component {
 		const path = this.props.siteBasePath;
 
 		// Filter out jetpack sites when on particular routes
-		if ( /^\/customize/.test( path ) ) {
+		if ( /^\/customize/.test( path ) || /^\/domains/.test( path ) ) {
 			return ! site.jetpack;
 		}
 
 		// Filter out sites with no upgrades on particular routes
-		if ( /^\/domains/.test( path ) || /^\/plans/.test( path ) ) {
```

Comment:
The rules are:
- `customize` route doesn't need to be filtered at all. It works for all sites. Some site types can be customized in Calypso, others redirect to WP Admin
- `domains` can be managed on Simple and Atomic sites, not Jetpack. The right filter condition is: `! site.jetpack || site.options.is_automated_transfer`
- `plans` condition is correct: `! site.jetpack || site.isSiteUpgradeable`